Documentation/class-libraries.md

```diff
@@ -0,0 +1,50 @@
+.NET Class Libraries
+====================
+
+Class libraries are the [shared library](http://en.wikipedia.org/wiki/Library_(computing)#Shared_libraries) concept for .NET. They enable you to componentize useful functionality into modules that can be used by multiple applications. They can also be used as a means of loading functionality that is not needed or not known at application startup. Class libraries are described using the [.NET Assembly file format](assembly-format.md).
+
+There are three types of class libraries that you can use:
+
+- **Platform-specific** class libraries have access to all the APIs in a given platform (e.g. .NET Framework, Xamarin iOS), but can only be used by apps and libraries that target that platform.
+- **Portable** class libraries have access to a subset of APIs, and can be used by apps and libraries that target multiple platforms.
+- **.NET Core** class libraries are a merger of the platform-specific and portable library concept into a single model that provides the best of both.
+
+Platform-specific Class Libraries
+=================================
+
+Platform-specific libraries are bound to a single .NET platform (e.g. .NET Framework on Windows) and can therefore take significant dependencies on a known execution environment. Such an environment will expose a known set of APIs (.NET and OS APIs) and will maintain and expose expected state (e.g. Windows registry). 
```

Comment:
I think my comment may have gone to the wrong line (or moved after a further change).  I meant for it to apply to the Portable Class Libraries description.  I think talking about a synthetic "known execution environment" confuses things there.  (I did suggest changing all the descriptions, but I think the PCL one is the important one).

In terms of the things you mentioned, from a PCL perspective whether those things are available maps to whether the APIs are available or not.  You can't access the registry because the registry APIs aren't portable.  You can use WinRT Xaml in portable libraries when the right APIs area available.